include/faker-cxx/string.h

```diff
@@ -120,9 +189,66 @@ std::string uuid(RandomGenerator<T> gen = RandomGenerator<std::mt19937>{})
     return result;
 }
 
-#ifdef __SIZEOF_INT128__
-#define ULIDUINT128
-#endif
+#pragma endregion
+
+/**
+ * @brief Generates an Universally Unique Identifier, defaults to V4.
+ *
+ * @param gen A random number generator (type RandomGenerator)
+ *
+ * @returns UUID.
+ *
+ * @code
+ * faker::string::uuid() // V4: "27666229-cedb-4a45-8018-98b1e1d921e2"
+ * @endcode
+ */
+template <typename T = std::mt19937>
+std::string uuid(RandomGenerator<T> gen)
+{
+    return uuid(Uuid::V4, gen);
+}
+
+/**
+ * @brief Generates an Universally Unique Identifier, defaults to V4.
+ *
+ * @param gen A random number generator (type RandomGenerator)
+ *
+ * @returns UUID.
+ *
+ * @code
+ * faker::string::uuid()         // "27666229-cedb-4a45-8018-98b1e1d921e2" // V4
+ * faker::string::uuid(Uuid::V1) // "04f916a0-af32-11ef-9cd2-0242ac120002"
+ * faker::string::uuid(Uuid::V3) // "a3bb189e-8bf9-3888-9912-ace4e6543002"
+ * faker::string::uuid(Uuid::V4) // "27666229-cedb-4a45-8018-98b1e1d921e2"
+ * faker::string::uuid(Uuid::V5) // "27666229-cedb-4a45-8018-98b1e1d921e2"
+ * faker::string::uuid(Uuid::V6) // "27666229-cedb-4a45-8018-98b1e1d921e2"
+ * faker::string::uuid(Uuid::V7) // "27666229-cedb-4a45-8018-98b1e1d921e2"
+ * faker::string::uuid(Uuid::V8) // "27666229-cedb-4a45-8018-98b1e1d921e2"
+ * @endcode
+ */
+template <typename T = std::mt19937>
+std::string uuid(Uuid uuid = Uuid::V4, RandomGenerator<T> gen = RandomGenerator<std::mt19937>{})
+{
+    switch (uuid)
+    {
+    case Uuid::V1:
+        return uuidV1(gen);
+    case Uuid::V3:
+        return uuidV3(gen);
+    case Uuid::V4:
+        return uuidV4(gen);
+    case Uuid::V5:
+        return uuidV4(gen);
+    case Uuid::V6:
+        return uuidV4(gen);
+    case Uuid::V7:
+        return uuidV4(gen);
+    case Uuid::V8:
+        return uuidV4(gen);
+    default:
+        return uuidV4(gen);
+    }
```

Comment:
move all implementation to cpp file

Reply:
They are templates, because of the generator, I can't move those without linker errors.

Reply:
So maybe we should not expose template params to end users and hide generator in implementation, what do you think about that?

Reply:
If i would use it I would like to jjst get uuid and not to pass and params ideally, so it could be generated inside

Reply:
I agree, there is no reason why the users would want to have their own generator, they will likely just use the default one.